core/rawdb/accessors_indexes.go

```diff
@@ -39,26 +40,38 @@ type TxLookupEntry struct {
 
 // ReadTxLookupEntry retrieves the positional metadata associated with a transaction
 // hash to allow retrieving the transaction or receipt by hash.
-func ReadTxLookupEntry(db kv.Getter, txnHash libcommon.Hash) (*uint64, error) {
+func ReadTxLookupEntry(db kv.Getter, txnHash libcommon.Hash) (*uint64, *uint64, error) {
 	data, err := db.GetOne(kv.TxLookup, txnHash.Bytes())
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	if len(data) == 0 {
-		return nil, nil
+		return nil, nil, nil
 	}
-	number := new(big.Int).SetBytes(data).Uint64()
-	return &number, nil
+	numberBlockNum := new(big.Int).SetBytes(data[:min(8, len(data))]).Uint64()
+
+	var numberTxNum uint64
+	if len(data) >= 8 {
+		numberTxNum = new(big.Int).SetBytes(data[8:]).Uint64()
+		numberTxNum += 2
```

Comment:
completely don't understand these magic number @AskAlexSharov maybe you know (#1)

Reply:
- `min(8, len(data))]` must be `8`.
- `if len(data) >= 8 {` add `else panic("bad data")` to fail-fast.
- `(*uint64, *uint64, error)` add names for return parameters. otherwise hard to understand what you return. right now i don't understand what is `numberTxNum`. 
- `+= 2` means some code which calling this func expecting something special from your func. break this expectations. who calling  `ReadTxLookupEntry` and using result txNum?